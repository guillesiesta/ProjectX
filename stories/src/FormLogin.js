import React, { Component } from 'react';

class FormLogin extends Component {

  constructor(props) {
      super(props);
      this.state = {username: '', password: ''};

      this.handleNameChange = this.handleNameChange.bind(this);
      this.handlePasswordChange = this.handlePasswordChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({username: event.target.value});
    }
    handlePasswordChange(event) {
        this.setState({password: event.target.value});
    }

    handleSubmit(event) {
      console.log('A name was submitted: ' + this.state.username + ' & A password was submitted: ' + this.state.password);
      this.props.getUsername(this.state.username);
      event.preventDefault();
    }

    render(){
      return(
        <form onSubmit={this.handleSubmit}>
          <label>User:</label>
             <input type="text" value={this.state.username} onChange={this.handleNameChange} />

          <label>Password:</label>
             <input type="password" value={this.state.password} onChange={this.handlePasswordChange} />

          <input type="submit" value="Submit" />
        </form>
      );
    }

}

export default FormLogin;
