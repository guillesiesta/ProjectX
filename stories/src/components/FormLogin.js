import React, { Component } from 'react';
import url from '../url/url'
//import axios from 'axios'

class FormLogin extends Component {

  constructor(props) {
      super(props);
      this.state = {username: '', password: ''};

      this.handleNameChange = this.handleNameChange.bind(this);
      this.handlePasswordChange = this.handlePasswordChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({username: event.target.value});
        var me = sessionStorage.getItem('username');
        console.log("GET local storage: "+ me);
    }
    handlePasswordChange(event) {
        this.setState({password: event.target.value});
    }

    handleSubmit(event) {
      console.log('A name was submitted: ' + this.state.username + ' & A password was submitted: ' + this.state.password);
      //this.props.getUsername(this.state.username); //aquÃ­ es donde hay que llamar si queremos que nos devuelva el nombre del usuario

      /*axios.get('http://localhost:5000/login',{usern:this.state.username, passw:this.state.password})
        .then(function(response){
          var data = response['data'];
          //console.log("DATA: "+JSON.stringify(data[0].nick));
          console.log("el nick es:"+ response.data[0].nick);
        });*/
        fetch('https://projectx-eagwiugnbd.now.sh/login', {
              method: 'POST', // or 'PUT'
              body: JSON.stringify(this.state.username), // data can be `string` or {object}!
              headers:{
                'Content-Type': 'application/json'
              }})
          .then(response => response.json())
          .then(data => {
            //console.log(data[0].nick) // Prints result from `response.json()` in getRequest
            //console.log(data[0].nick)
            //console.log(data[0].password)
            //console.log(data)
            if(this.state.password=== data[0].password){
              this.props.getUsername(this.state.username);
            }
          })
          .catch(error => console.error(error))
/*
        axios.get('http://localhost:5000/login')
          .then(response => console.log("ESTO: " +response.data.storie+"+ ESto:" +response.data.title));
*/
      event.preventDefault();
    }

    render(){
      return(
        <form onSubmit={this.handleSubmit}>
          <label>User:</label>
             <input type="text" value={this.state.username} onChange={this.handleNameChange} />

          <label>Password:</label>
             <input type="password" value={this.state.password} onChange={this.handlePasswordChange} />

          <input type="submit" value="Entrar" />
        </form>
      );
    }

}

export default FormLogin;
