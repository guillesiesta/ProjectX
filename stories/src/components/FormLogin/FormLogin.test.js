import React, {Component} from 'react';
import { shallow } from 'enzyme';

import FormLogin from './FormLogin';

describe('FormLogin', ()=> {
  test('Rendering the simple component', ()=>{

    const wrapper = shallow(
      <FormLogin getUsername="usuario"/>
    );

    expect(wrapper).toMatchSnapshot();
  })
})
