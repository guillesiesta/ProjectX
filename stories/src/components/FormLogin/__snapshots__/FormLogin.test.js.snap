// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin Rendering the simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLogin
    getUsername="usuario"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-md-4 col-md-offset-4"
        >
          <form
            onSubmit={[Function]}
          >
            <img
              alt="logo"
              className="img-rounded"
              src="img/logo_riddling.png"
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Usuario"
              type="text"
              value=""
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
            <p />
            <input
              className="col-md-5 col-md-offset-1"
              type="submit"
              value="Entrar"
            />
          </form>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-4 col-md-offset-4"
        >
          <form
            onSubmit={[Function]}
          >
            <img
              alt="logo"
              className="img-rounded"
              src="img/logo_riddling.png"
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Usuario"
              type="text"
              value=""
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
            <p />
            <input
              className="col-md-5 col-md-offset-1"
              type="submit"
              value="Entrar"
            />
          </form>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
          >
            <img
              alt="logo"
              className="img-rounded"
              src="img/logo_riddling.png"
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Usuario"
              type="text"
              value=""
            />
            <p />
            <input
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
            <p />
            <input
              className="col-md-5 col-md-offset-1"
              type="submit"
              value="Entrar"
            />
          </form>,
          "className": "col-md-4 col-md-offset-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="img-rounded"
                src="img/logo_riddling.png"
              />,
              <p />,
              <input
                onChange={[Function]}
                placeholder="Usuario"
                type="text"
                value=""
              />,
              <p />,
              <input
                onChange={[Function]}
                placeholder="Contraseña"
                type="password"
                value=""
              />,
              <p />,
              <input
                className="col-md-5 col-md-offset-1"
                type="submit"
                value="Entrar"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "img-rounded",
                "src": "img/logo_riddling.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Usuario",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Contraseña",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-md-5 col-md-offset-1",
                "type": "submit",
                "value": "Entrar",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-md-4 col-md-offset-4"
          >
            <form
              onSubmit={[Function]}
            >
              <img
                alt="logo"
                className="img-rounded"
                src="img/logo_riddling.png"
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Usuario"
                type="text"
                value=""
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Contraseña"
                type="password"
                value=""
              />
              <p />
              <input
                className="col-md-5 col-md-offset-1"
                type="submit"
                value="Entrar"
              />
            </form>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-md-4 col-md-offset-4"
          >
            <form
              onSubmit={[Function]}
            >
              <img
                alt="logo"
                className="img-rounded"
                src="img/logo_riddling.png"
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Usuario"
                type="text"
                value=""
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Contraseña"
                type="password"
                value=""
              />
              <p />
              <input
                className="col-md-5 col-md-offset-1"
                type="submit"
                value="Entrar"
              />
            </form>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <img
                alt="logo"
                className="img-rounded"
                src="img/logo_riddling.png"
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Usuario"
                type="text"
                value=""
              />
              <p />
              <input
                onChange={[Function]}
                placeholder="Contraseña"
                type="password"
                value=""
              />
              <p />
              <input
                className="col-md-5 col-md-offset-1"
                type="submit"
                value="Entrar"
              />
            </form>,
            "className": "col-md-4 col-md-offset-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="img-rounded"
                  src="img/logo_riddling.png"
                />,
                <p />,
                <input
                  onChange={[Function]}
                  placeholder="Usuario"
                  type="text"
                  value=""
                />,
                <p />,
                <input
                  onChange={[Function]}
                  placeholder="Contraseña"
                  type="password"
                  value=""
                />,
                <p />,
                <input
                  className="col-md-5 col-md-offset-1"
                  type="submit"
                  value="Entrar"
                />,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "img-rounded",
                  "src": "img/logo_riddling.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Usuario",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Contraseña",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "col-md-5 col-md-offset-1",
                  "type": "submit",
                  "value": "Entrar",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
